File: mobile-app/apps/edulynk/scripts/bake-edulynk-icons.js
/**
 * Bake Edulynk mark into Expo + Play Store icon slots.
 * Source: assets/edulynk-mark-source.png (official mark on black).
 */
const path = require('path');
const sharp = require('sharp');

const ASSETS = path.join(__dirname, '..', 'assets');
const PLAY = path.join(ASSETS, 'play-store');
const SOURCE = path.join(ASSETS, 'edulynk-mark-source.png');
const NAVY = { r: 7, g: 26, b: 61, alpha: 1 };

async function bakeSquare(size, outPath, { padRatio = 0.12 } = {}) {
  const pad = Math.round(size * padRatio);
  const inner = size - pad * 2;
  const mark = await sharp(SOURCE)
    .resize(inner, inner, { fit: 'contain', background: NAVY })
    .png()
    .toBuffer();

  await sharp({
    create: {
      width: size,
      height: size,
      channels: 4,
      background: NAVY,
    },
  })
    .composite([{ input: mark, left: pad, top: pad }])
    .png()
    .toFile(outPath);

  console.log('wrote', path.basename(outPath), `${size}x${size}`);
}

async function bakeSplash() {
  const size = 1024;
  const markSize = 720;
  const mark = await sharp(SOURCE)
    .resize(markSize, markSize, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
    .png()
    .toBuffer();
  const left = Math.round((size - markSize) / 2);
  await sharp({
    create: {
      width: size,
      height: size,
      channels: 4,
      background: { r: 0, g: 0, b: 0, alpha: 0 },
    },
  })
    .composite([{ input: mark, left, top: left }])
    .png()
    .toFile(path.join(ASSETS, 'splash-icon.png'));
  console.log('wrote splash-icon.png');
}

async function main() {
  await bakeSquare(1024, path.join(ASSETS, 'icon.png'), { padRatio: 0.08 });
  await bakeSquare(1024, path.join(ASSETS, 'adaptive-icon.png'), { padRatio: 0.18 });
  await bakeSplash();
  await bakeSquare(512, path.join(PLAY, 'app-icon-512x512.png'), { padRatio: 0.08 });
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: mobile-app/apps/edulynk/scripts/bake-edulynk-icons.py
"""Bake the official Edulynk mark into Expo + Play Store icon slots.

Prefer `bake-edulynk-icons.js` (Node + sharp). This script is a Pillow fallback.
"""
from __future__ import annotations

from pathlib import Path

from PIL import Image

ROOT = Path(__file__).resolve().parents[1]
ASSETS = ROOT / "assets"
PLAY = ASSETS / "play-store"
SOURCE = ASSETS / "edulynk-mark-source.png"
NAVY = (7, 26, 61, 255)


def bake_square(size: int, out: Path, pad_ratio: float = 0.12) -> None:
    pad = round(size * pad_ratio)
    inner = size - pad * 2
    src = Image.open(SOURCE).convert("RGBA")
    mark = src.resize((inner, inner), Image.Resampling.LANCZOS)
    canvas = Image.new("RGBA", (size, size), NAVY)
    canvas.paste(mark, (pad, pad), mark)
    canvas.convert("RGB").save(out, "PNG", optimize=True)
    print(f"wrote {out.name} {size}x{size}")


def bake_splash() -> None:
    size = 1024
    mark_size = 720
    src = Image.open(SOURCE).convert("RGBA")
    mark = src.resize((mark_size, mark_size), Image.Resampling.LANCZOS)
    canvas = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    offset = (size - mark_size) // 2
    canvas.paste(mark, (offset, offset), mark)
    canvas.save(ASSETS / "splash-icon.png", "PNG", optimize=True)
    print("wrote splash-icon.png")


def main() -> None:
    PLAY.mkdir(parents=True, exist_ok=True)
    bake_square(1024, ASSETS / "icon.png", 0.08)
    bake_square(1024, ASSETS / "adaptive-icon.png", 0.18)
    bake_splash()
    bake_square(512, PLAY / "app-icon-512x512.png", 0.08)


if __name__ == "__main__":
    main()
